notifications/src/main/kotlin/com/d3/notifications/init/NotificationInitialization.kt

```diff
@@ -63,11 +69,20 @@ class NotificationInitialization(
      * @param block - block to handle
      */
     private fun handleBlock(block: BlockOuterClass.Block) {
-        getSetDetailCommands(block).map { it.setAccountDetail }
-            .forEach { setDetailCommand ->
+        getSetDetailCommandsWithCreator(block)
+            .forEach { commandWithCreator ->
+                val command = commandWithCreator.command.setAccountDetail
                 // Notify withdrawal
-                if (isWithdrawal(setDetailCommand)) {
-                    handleWithdrawalEventNotification(setDetailCommand)
+                if (isWithdrawal(command)) {
```

Comment:
replace these ifs with some sort of handler in a chain of responsibilities fashion?

Reply:
It's a good idea, but it might take a while. I have a special todo at the top of the file `//TODO refactor handlers`